File: Questao4.py
Faturamento_Estados = {
    "SP": 67836.43,
    "RJ": 36678.66,
    "MG": 29229.88,
    "ES": 27165.48,
    "Outros": 19849.53
}

# Obtendo valor total
valor_total = sum(Faturamento_Estados.values())

# Rodando um laco para calcular o percentual de cada estado e exibi-lô
for estado, valor in Faturamento_Estados.items():
    percentual = (valor / valor_total) * 100
    print(f"Estado: {estado}, Porcentagem: {percentual:.2f}%")

File: Questao4.ts
// # Faturamento estado

const faturamentoEstados: { estado: string; valor: number }[] = [
  { estado: "SP", valor: 67836.43 },
  { estado: "RJ", valor: 36678.66 },
  { estado: "MG", valor: 29229.88 },
  { estado: "ES", valor: 27165.48 },
  { estado: "Outros", valor: 19849.53 },
];

function calcularPercentual(faturamento: { estado: string; valor: number }[]) {
  const valorTotal: number = faturamento.reduce(
    (total, estado) => total + estado.valor,
    0
  );

  return faturamento.map((estado) => {
    return {
      estado: estado.estado,
      percentual: (estado.valor / valorTotal) * 100,
    };
  });
}

const percentuais = calcularPercentual(faturamentoEstados);
percentuais.forEach((estado) => {
  console.log(
    `Estado: ${estado.estado}, Percentual: ${estado.percentual.toFixed(2)}%`
  );
});
